### DISCUSSION TITLE
Avaliem a qualidade do meu código(HTML/CSS/JS), por favor?

### DISCUSSION THREAD

**Discussion by vitorvd** [frontendbr/forum#1957] [Dicas]
Oi amigxs, tudo bem? Espero que sim!

Comecei levar a sério o desenvolvimento web há uma semana e tentei centralizar todos os conhecimentos adquiridos ao longo dessa pequena jornada em um só web-site(https://github.com/vitorvd/vitorvd-project), fiz ele com intuito de ser um portfólio no futuro. 
Venho aqui, pedir encarecidamente a vocês me passem um feedbeck em relação ao meu código, dicas, sugestões e ideias para melhorar o site mesmo.

Link para o repositório: https://github.com/vitorvd/vitorvd-project

#Features: 
- HTML 5; 
- CSS 3; 
- JavaScript (para desenvolver o menu hambúrguer). 
- O site é totalmente responsivo com Flexbox e CSS Grid, uso também o pré compilador: SASS.

Deixo aqui meu agradecimento a todos que contribuírem.

**Comment by brunopulis** [comment]
Fala @vitorvd  tudo bem? 

Eu avaliei somente a estrutura da sua página vou contribuir com alguns pontos de possíveis melhorias. 

## Ferramentas

Para fazer essa avaliação eu utilizei o [validador de HTML da W3C](https://validator.w3.org/nu) e o [Access Monitor](https://accessmonitor.acessibilidade.gov.pt/)

## Recomendações: 

### Declarar o idioma principal da página

Declarar o idioma da página (auxilia tradutores como Google Translate e leitores de tela), você pode aprofundar no [assunto](https://webdesign.tutsplus.com/tutorials/use-the-lang-attribute-for-better-accessibility--cms-31961); 

A dica aqui e na tag `html` incluir o atributo com a linguagem definida para o site, por exemplo, se o conteúdo do seu site for totalmente em português.

Exemplo: 

```html
<html lang="pt-br">
```

Um detalhe que percebi você utiliza no menu principal, alguns termos em inglês, então seria prudente colocar o indicativo dessas palavras em inglês. Ficaria assim: 

```html
<nav>
      <!-- menu default -->
      <ul class="menu-desktop">
          <li><a href="hi.html"><span lang="en">hi</span></a></li>
          <li><a href="#"><span lang="en">jobs</span></a></li>
          <li><a href="#"><span lang="en">contact</span></a></li>
      </ul>
      <!-- só será exibido quando clicado no menu hamburger -->
      <ul class="menu-mobile">
          <li><a href="hi.html"><span lang="en">hi</span></a></li>
          <li><a href="#"><span lang="en">jobs</span></a></li>
          <li><a href="#"><span lang="en">contact</span></a></li>
          <li><i class="menu-link fas fa-times"></i></li>
      </ul>
      <i class="menu-hamburguer fas fa-bars"></i>
</nav>
```

### Retirar atributo type da tag <script> 

Com o advento do HTML5, não se faz necessário usar o atributo `type` mais nas tags `<script>`

### Uso do Font Awesome 

Aqui é mais um macete opte usar `<span>` ou invés da tag `<i>`, [nesse artigo](https://brunopulis.com/drops/font-awesome-semantico/) que explico o motivo.

### Sections

Procure utlizar headings entre `h2 a h6` não é recomendável usar `<h1>` sobre o assunto você pode [consultar aqui](https://www.w3.org/WAI/WCAG21/Techniques/general/G141.html).

### Cards de Portfólio 

Uma curiosidade, pessoas que usam um leitor de telas podem navegar através de links, com a tecla "L" do teclado ela irá pular de link a link na página. 

Ao chegar no link "Ver aplicação" ele não saberá para onde esse link direciona, nesse caso você pode fazer duas coisas:

* descrever melhor o link;
* usar um recurso de texto oculto. 

#### Descrever melhor o link 

Talvez pensar em uma forma que descreva o conteúdo seja uma boa opção.

#### Texto oculto 

Você pode fazer algo, como: 

```html
<div class="card-text">
     <!-- considerando que você utilizei no titulo da section um h2  -->
      <h3>Portfólio</h3>
      <p>Meu primeiro website profissional, feito com paciência, muito estudo e carinho. 
      A única finalidade é de aprendizado, desta forma, o torno open source!</p>
      <a href="#">Ver aplicação <span class="sr-only">do meu site</span></a>
</div>
```

### Descrever os links no Footer 

Uma boa prática é descrever os links do rodapé do site. As redes sociais da forma que você desenvolveu se o script do Font Awesome não carregar por algum motivo, ele ficará em branco e sem nenhuma informação. 

Recomendaria você dar uma pesquisa sobre uma classe chamada `sr-only` do Bootstrap, ela oculta visualmente uma informação mas o conteúdo ainda fica no seu HTML, isso auxilia pessoas que não enxergar conseguir encontrar as informações no seu site. 

Um exemplo: 

```html
<footer>
        <div>
            <a href="https://twitter.com/_vitorvd" target="_blank" rel="noopener noreferrer">
               <span class="fab fa-twitter" aria-hidden="true"></span>
               <span class="sr-only">Twitter (abre em uma nova aba)</span>
            </a>
            <a href="https://github.com/vitorvd" target="_blank" rel="noopener noreferrer">
                <span class="fab fa-github" aria-hidden="true"></span>
                <span class="sr-only">Github (abre em uma nova aba)</span>
            </a>
            <a href="https://www.linkedin.com/in/vitor-versiani-8b5b05150/" target="_blank" rel="noopener noreferrer">
               <span class="fab fa-linkedin-in" aria-hidden="true"></span>
               <span class="sr-only">Github (abre em uma nova aba)</span>
             </a>
        </div>
</footer>
``` 

Espero que tenha ficado claro, qualquer dúvida só chamar =)

**Reply by vitorvd** [reply]
Valeu pelas dicas ;)
